src/main/java/org/eclipse/xpanse/terraform/boot/terraform/service/TerraformTaskResultHelper.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ */
+
+package org.eclipse.xpanse.terraform.boot.terraform.service;
+
+import jakarta.annotation.Resource;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.serializer.Serializer;
+import org.eclipse.xpanse.terraform.boot.models.exceptions.FailedCallbackResponseStoreException;
+import org.eclipse.xpanse.terraform.boot.models.exceptions.ResultAlreadyReturnedOrRequestIdInvalidException;
+import org.eclipse.xpanse.terraform.boot.models.response.TerraformResult;
+import org.eclipse.xpanse.terraform.boot.utils.SerializerUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+/**
+ * Terraform service classes are manage task result.
+ */
+@Slf4j
+@Component
+public class TerraformTaskResultHelper {
+
+    private static final String TF_RESULT_FILE_SUFFIX = ".dat";
+
+    @Value("${failed.callback.response.store.location}")
+    private String failedCallbackStoreLocation;
+
+    @Resource
+    private SerializerUtils serializerUtils;
+
+    /**
+     * When the terraform-boot callback fails, store the TerraformResult in the local file system.
+     *
+     * @param result TerraformResult.
+     */
+    public void storeFailedCallbackResponse(TerraformResult result) {
+        String filePath = getFilePath(result.getRequestId());
+        File file = new File(filePath);
+        if (!file.exists() && !file.mkdirs()) {
+            throw new FailedCallbackResponseStoreException("Failed to create directory: "
+                    + filePath);
+        }
+        Serializer<byte[]> serializer = serializerUtils
```

Comment:
we must move this entire logic to the SerializerUtils. 

Just send the TerraformResult to 'serialize' method in SerializerUtil and get the byte[] data.